modules/nextflow/src/main/groovy/nextflow/datasource/SraExplorer.groovy

```diff
@@ -233,18 +253,22 @@ class SraExplorer {
         getCacheFolder().resolve("$bucket/${acc}.fastq_ftp.cache")
     }
 
-    protected String readRunFastqs(String acc) {
+    protected String readRunFastqs(String acc, List<String> queryFields) {
         final Path cache = cachePath(acc)
         if( useCache ) try {
             def delta = System.currentTimeMillis() - FilesEx.lastModified(cache)
-            if( delta < CACHE_MAX_TIME.millis )
-                return cache.getText()
+            if( delta < CACHE_MAX_TIME.millis ) {
+                def text = cache.getText()
+                def cached = matchCache(text, queryFields)
+                if (cached)
+                    return text
```

Comment:
Why this?

Reply:
Since it could exist a cache that was generated using the same ID but not the same fields and then it was returning the same fields as the original run without re-running it.